boa_examples/src/bin/module_fetch.rs

```diff
@@ -0,0 +1,258 @@
+struct HttpModuleLoader;
+
+impl ModuleLoader for HttpModuleLoader {
+    fn load_imported_module(
+        &self,
+        _referrer: boa_engine::module::Referrer,
+        specifier: JsString,
+        finish_load: Box<dyn FnOnce(boa_engine::JsResult<Module>, &mut Context<'_>)>,
+        context: &mut Context<'_>,
+    ) {
+        let url = specifier.to_std_string_escaped();
+
+        let fetch = async move {
+            // `surf` requires to manually specify following redirects.
+            let client = Client::default().with(Redirect::default());
+
+            // Adding some prints to show the non-deterministic nature of the async fetches.
+            // Try to run the example several times to see how sometimes the fetches start in order
+            // but finish in disorder.
+            println!("Fetching `{url}`...");
+            // This could also retry fetching in case there's an error while requesting the module.
+            let body = client.get(&url).recv_string().await;
+            println!("Finished fetching `{url}`!");
+
+            // Since the async context cannot take the `context` by ref, we have to continue
+            // parsing inside a new `NativeJob` that will be enqueued into the promise job queue.
+            NativeJob::new(move |context| -> JsResult<JsValue> {
+                let body = match body {
+                    Ok(body) => body,
+                    Err(err) => {
+                        // On error we always call `finish_load` to notify the load promise about the
+                        // error.
+                        finish_load(
+                            Err(JsNativeError::typ().with_message(err.to_string()).into()),
+                            context,
+                        );
+
+                        // Just returns anything to comply with `NativeJob::new`'s signature.
+                        return Ok(JsValue::undefined());
+                    }
+                };
+
+                // Could also add a path if needed.
+                let source = Source::from_bytes(body.as_bytes());
+
+                let module = Module::parse(source, None, context);
+
+                // We don't do any error handling, `finish_load` takes care of that for us.
+                finish_load(module, context);
+
+                // Also needed to match `NativeJob::new`.
+                Ok(JsValue::undefined())
+            })
+        };
+
+        // Just enqueue the future for now. We'll advance all the enqueued futures inside our custom
+        // `JobQueue`.
+        context
+            .job_queue()
+            .enqueue_future_job(Box::pin(fetch), context)
+    }
+}
+
+fn main() -> JsResult<()> {
+    // A simple snippet that imports modules from the web instead of the file system.
+    const SRC: &str = r#"
+        import YAML from 'https://esm.run/yaml';
+        import fromAsync from 'https://esm.run/array-from-async';
+        import { Base64 } from 'https://esm.run/js-base64';
```

Comment:
Should we specify a version here? To ensure that It doesn't break it future releases.
```suggestion
        import YAML from 'https://esm.run/yaml@2.3.1';
        import fromAsync from 'https://esm.run/array-from-async@3.0.0';
        import { Base64 } from 'https://esm.run/js-base64@3.7.5';
```